src/Domain/Model/CreditCardPayment.php

```diff
@@ -39,6 +39,6 @@ public function getValuationDate(): ?DateTimeImmutable {
 	}
 
 	public function paymentCompleted(): bool {
-		return $this->creditCardData !== null;
+		return $this->creditCardData->getTransactionId() !== '';
```

Comment:
Since this can at least in theory be nullable, could you do

```suggestion
		return $this->creditCardData !== null && $this->creditCardData->getTransactionId() !== '';
```

instead?

Reply:
Whoops, didn't see it could be set null in the constructor. So it should be checked or it'll throw an exception